### PR TITLE
add NOTICE, licenses-report, license headers

### PR DESCRIPTION
**Pull request by csplinter**

### CODE REVIEW

File: cql/src/main/java/org/apache/cassandra/stargate/cql3/DefaultQueryOptions.java [hunk 1]
```diff
@@ -1,3 +1,18 @@
+/*
```
Comment (dougwettlaufer) [inline cql/src/main/java/org/apache/cassandra/stargate/cql3/DefaultQueryOptions.java:1]: I don't know about this one. Anything under org.apache.cassandra is probably borrowed. @mpenick ?

File: licenses-report.txt [hunk 2]
```diff
@@ -0,0 +1,351 @@
+# 3rd-Party Software License Notice
```
Comment (dougwettlaufer) [inline licenses-report.txt:1]: Is there a maven plugin that generates this on the fly?

Comment (dougwettlaufer) [reply licenses-report.txt:1]: We should probably add something to CI to make sure it's ran too.

File: persistence-api/src/main/java/io/stargate/db/datastore/schema/Keyspace.java [hunk 3]
```diff
@@ -32,7 +32,7 @@
 
     /**
      * This list is verified in a test against SystemInfo. We don't want to use this directly though for
-     * ExternalDataStore as it shouldn't have dependencies on DSE internal classes
+     * ExternalDataStore as it shouldn't have dependencies on Cassandra's internal classes
```
Comment (dougwettlaufer) [inline persistence-api/src/main/java/io/stargate/db/datastore/schema/Keyspace.java:35]: What about the header on this one?

File: licenses-report.txt (hunk 2 above)
Comment (csplinter) [reply licenses-report.txt:1]: @olim7t mentioned that this is what the Java driver is using https://github.com/datastax/java-driver/blob/4.x/pom.xml#L622-L664 

Not sure how we'd configure it differently based on the files

File: persistence-api/src/main/java/io/stargate/db/datastore/schema/Keyspace.java (hunk 3 above)
Comment (csplinter) [reply persistence-api/src/main/java/io/stargate/db/datastore/schema/Keyspace.java:35]: Talked to Brittni from Legal, she said to leave it for now while she circles w/Jason. Will wait to hear back. Did we modify any of the DSE code?

File: auth-api/src/main/java/io/stargate/auth/api/AuthApiActivator.java [hunk 4]
```diff
@@ -1,3 +1,18 @@
+/*
+ * Copyright The Stargate Authors
```
Comment (dimas-b) [inline auth-api/src/main/java/io/stargate/auth/api/AuthApiActivator.java:2]: Do we use this header in all sub-modules or just in `stargate/stargate`?

Comment (dimas-b) [reply auth-api/src/main/java/io/stargate/auth/api/AuthApiActivator.java:2]: Do we want the copyright header in `pom.xml` files?

Comment (dougwettlaufer) [reply auth-api/src/main/java/io/stargate/auth/api/AuthApiActivator.java:2]: All repos

File: persistence-api/src/main/java/io/stargate/db/datastore/schema/Keyspace.java (hunk 3 above)
Comment (dougwettlaufer) [reply persistence-api/src/main/java/io/stargate/db/datastore/schema/Keyspace.java:35]: For the most part, no.

File: licenses-report.txt (hunk 2 above)
Comment (olim7t) [reply licenses-report.txt:1]: The plugin supports inclusion/exclusion filters. We can have multiple executions that operate on specific directories.

**Comment by dougwettlaufer** [comment]
Sorry, merge conflicts.

**Review by dougwettlaufer — APPROVED** [review]

**Pull request merged**